### ISSUE TITLE
Deduplication strategy None does not always work as expected when running load tests

### ISSUE THREAD

**Issue by bastbu** [Azure/iotedge-lorawan-starterkit#1361]
## Expected Behavior

When running load tests (e.g. test `Connected_Factory_Load_Test_Scenario` on 5d8b1267997e539c3bd97278c921aa44ac24595f), for the parameters: 
- number of factories: 1
- number of concentrators: 2
- number of devices: 4
- number of gateways: 2
- deduplication strategy: None

Some messages are delivered to IoT Hub twice, even though they should be delivered only once (each concentrator per factory is connected to a different gateway).

## Current Behavior

Sometimes, six messages instead of five arrive at IoT Hub.

**Comment by bastbu** [comment]
Closing as this is already tracked in #1032
